src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -560,6 +497,22 @@ out processInfo
 #endif
         }
 
+        /// <summary>
+        /// Identify the .NET host of the current process
+        /// </summary>
+        /// <returns>The full path to the executable hosting the current process, or null if running on Full Framework on Windows.</returns>
+        private static string GetCurrentHost()
```

Comment:
I think there is an environment variable set by dotnet.exe which we can read.  I think this method should return the path to dotnet.exe or msbuild.exe and not null...